server/src/main/scala/com/cloudera/livy/LivyConf.scala

```diff
@@ -246,14 +246,20 @@ class LivyConf(loadDefaults: Boolean) extends ClientConf[LivyConf](null) {
   def sparkDeployMode(): Option[String] = Option(get(LIVY_SPARK_DEPLOY_MODE)).filterNot(_.isEmpty)
 
   /** Return the location of the spark home directory */
-  def sparkHome(): Option[String] = Option(get(SPARK_HOME)).orElse(sys.env.get("SPARK_HOME"))
+  def sparkHome(version: Option[String] = None): Option[String] = {
+    version.map {version => Option(get(s"livy.server.spark-home_$version"))
```

Comment:
I think we should abstract out `livy.server.spark-home_$version` here into a def like the val's above. Something along the lines of:
`def SPARK_HOME_VER(version: String) = Entry(s"livy.server.spark-home-$version", null)`